hexa/pipelines/schema/mutations.py

```diff
@@ -43,11 +44,35 @@ def resolve_create_pipeline(_, info, **kwargs):
         }
 
     try:
-        pipeline = Pipeline.objects.create(
-            code=input["code"],
-            name=input.get("name"),
-            workspace=workspace,
-        )
+        notebook = input.get("notebookPath")
+        if notebook:
+            # we need to check if the notebook path exist in the workspace bucket
+            get_bucket_object(workspace.bucket_name, notebook)
+            pipeline = Pipeline.objects.create(
+                code=input["code"],
+                name=input.get("name"),
+                workspace=workspace,
+                type=PipelineType.NOTEBOOK,
+                notebookPath=notebook,
+            )
+            # automatically create a new version
+            pipeline.upload_new_version(
+                user=request.user,
+                name=input.get("name"),
+                zipfile=base64.b64decode("".encode("ascii")),
```

Comment:
I thought you would remove this by setting the field `zipfile` as non mandatory ?